### PR TITLE
Use androidx screenshots to support newer android versions

### PR DESCRIPTION
**Pull request by Zemotacqy**
We want to support androidx screenshots for newer android versions. These changes would help users on working examples on how to use androidx screenshots in espresso tests.

### CODE REVIEW

File: app/src/androidTest/java/com/sample/browserstack/samplecalculator/ScreenshotUtils.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    public void captureScreenshot(String screenshotName) {
+        ScreenCapture capture = Screenshot.capture();
+        capture.setFormat(Bitmap.CompressFormat.PNG);
+        capture.setName(screenshotName);
+
+        try {
+            customProcesssor.process(capture);
+        } catch (IOException e) {
```
Comment (gshah30) [inline app/src/androidTest/java/com/sample/browserstack/samplecalculator/ScreenshotUtils.java:36]: this should `error` the test

File: app/src/androidTest/java/com/sample/browserstack/samplecalculator/EnsureInputTests.java [hunk 2]
```diff
@@ -39,18 +37,24 @@ public void setUp() {
 
     @Test
     public void ensureSingleInputIsHandled() {
-        Spoon.screenshot(mainActivity, "initial_state");
+        ScreenshotUtils screenshotUtils = new ScreenshotUtils();
+        screenshotUtils.captureScreenshot("initial_state");
+
         onView(withId(R.id.buttonOne)).perform(click());
         onView(withId(R.id.editText)).check(matches(withText("1")));
-        Spoon.screenshot(mainActivity, "post_multiple_btn_click");
```
[the remark this answers is not in the document]
Comment (Zemotacqy) [reply app/src/androidTest/java/com/sample/browserstack/samplecalculator/EnsureInputTests.java:45]: We do not need activity as we are not depending on activity to get a dump of window on screen which is the case with spoon. It does not uses androidx native libs.

File: app/src/androidTest/java/com/sample/browserstack/samplecalculator/ScreenshotUtils.java (hunk 1 above)
Comment (Zemotacqy) [reply app/src/androidTest/java/com/sample/browserstack/samplecalculator/ScreenshotUtils.java]: Yes, added the `throw new Exception` logic in the catch block which will ensure the same.

File: app/src/androidTest/java/com/sample/browserstack/samplecalculator/EnsureOperationTests.java [hunk 3]
```diff
@@ -70,7 +76,8 @@ public void ensureMultiplicationWorks() {
         onView(withId(R.id.buttonFive)).perform(click());
         onView(withId(R.id.buttonEqual)).perform(click());
         onView(withId(R.id.editText)).check(matches(withText("60")));
-        Spoon.screenshot(mainActivity, "post_multiplication");
+
```
Comment (Zemotacqy) [inline app/src/androidTest/java/com/sample/browserstack/samplecalculator/EnsureOperationTests.java:79]: Add comment that its an example.

Comment (Zemotacqy) [reply app/src/androidTest/java/com/sample/browserstack/samplecalculator/EnsureOperationTests.java:79]: Remove these kind of examples of nesting.

File: app/src/androidTest/java/com/sample/browserstack/samplecalculator/NativeScreenshot.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+public final class NativeScreenshot {
+
+    private static String methodName;
+    private static String className;
+    private static final Pattern SCREENSHOT_NAME_VALIDATION = Pattern.compile("[a-zA-Z0-9_-]+");
+
+    private NativeScreenshot() {}
+
+    /**
+     * Captures screenshot using Android Screenshot library and stores in the filesystem.
```
Comment (Zemotacqy) [inline app/src/androidTest/java/com/sample/browserstack/samplecalculator/NativeScreenshot.java:24]:
```suggestion
     * Captures screenshot using Androidx Screenshot library and stores in the filesystem.
```

**Review by gshah30 — APPROVED** [review]

**Review by nitish-bhardwaj-code — APPROVED** [review]

**Pull request merged**